drivers/can/Kconfig.nxp_s32

```diff
@@ -10,17 +10,23 @@ config CAN_NXP_S32_CANXL
 	  Enable support for NXP S32 CANXL driver.
 
 if CAN_NXP_S32_CANXL
+config CAN_NXP_S32_RX_FIFO
+	bool "NXP S32 CANXL uses RX FIFO"
+	default y
```

Comment:
Just for my understanding:
- Turning this off (`CONFIG_CAN_NXP_S32_RX_FIFO=n`) will make limit the driver to classic CAN?
- The rationale for introducing a Kconfig for this is to allow for more RX filters when not using CAN FD?

Reply:
> Just for my understanding:
> 
> * Turning this off (`CONFIG_CAN_NXP_S32_RX_FIFO=n`) will make limit the driver to classic CAN?

Yes, the limitation driver is not ability support to send CAN FD frame, just send CAN classic frame in both CAN classic and CAN FD mode (when remove CAN FD filtering)

> * The rationale for introducing a Kconfig for this is to allow for more RX filters when not using CAN FD?

This is not reason for introducing this kconfig. Actually, there are 32 RX filters for RXFIFO less than ones available for non-RXFIFO. The important reason for introducing, resolve the above limitation so that driver supports to send both CAN classic and CAN FD frames

Reply:
Okay, thanks for explaining this. 

Can I cherry-pick your commit into https://github.com/zephyrproject-rtos/zephyr/pull/65108 ?

Reply:
@henrikbrixandersen, what do you think that if I don't include your commits in this PR and open review PR independently as a new feature support? and you also should open a review PR and can merge first

Reply:
> Okay, thanks for explaining this.
> 
> Can I cherry-pick your commit into #65108 ?

If you prefer squash them into a PR. you can cherry-pick them

Reply:
> Hello @henrikbrixandersen, what do you think that if I don't include your commits in this PR and open review PR independently as a new feature support?

That would be great, thank. After your PR is merged, I will rebase on top of it.